pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -397,10 +427,17 @@ cdef class _Timestamp(ABCTimestamp):
         >>> ts.is_month_start
         True
         """
-        if self.freq is None:
+        if self._freq is None:
             # fast-path for non-business frequencies
             return self.day == 1
-        return self._get_start_end_field("is_month_start")
+        if self._needs_field_deprecation_warning(self._freq):
```

Comment:
might be worthile to just wrap this directly e.g.
```
self._warn_on_field_deprecation(self._freq, 'is_month_start')
```

then dont' have to repeat this everywhere

Reply:
good idea, updated